### PR TITLE
script to restart nexus

### PR DESCRIPTION
**Pull request by arunjadhavt**

### CODE REVIEW

File: site/profile/manifests/nexus/nexus_mem_check.pp [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# this is class to check nexus memory and restart nexus service if memory is below 500mb and other two instances under elb are in-service
+
+class profile::nexus::nexus_mem_check(
+  $nexus_cron_hours = '*',
+  $nexus_cron_minute = '*/10'
```
Comment (wanix) [inline site/profile/manifests/nexus/nexus_mem_check.pp]: this way to do is a bit hazardous, you can have 3 Nexus in low memory at the same time and all see we have 3 instances up and restart at the same time.
We must have different time check per node (offset).
With a inline_template calculation with 3mn offset, we should have as result:
NodeA: minute='0,10,20,30,40,50'
NodeB: minute='3,13,23,33,43,53'
NodeC: minute='6,16,26,36,46,56'

File: site/profile/manifests/nexus/nexus_mem_check.pp [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+){
+  if hiera('profile::nexus_restart_cron::enable'){
```
Comment (wanix) [inline site/profile/manifests/nexus/nexus_mem_check.pp]: not the good way to do.
you should have a parameter for this class:
```
class profile::nexus::nexus_mem_check(
  $nexus_cron_hours = '*',
  $nexus_cron_minute = '*/10',
  $enabled = false,
```
Don't forget to add `profile::nexus::nexus_mem_check::enabled: true` in vagrant hiera file for your tests

File: site/profile/manifests/nexus/nexus_mem_check.pp (hunk 1 above)
Comment (arunjadhavt) [reply site/profile/manifests/nexus/nexus_mem_check.pp]: I will check and time as per nodes

File: site/profile/manifests/nexus/nexus_mem_check.pp (hunk 2 above)
Comment (arunjadhavt) [reply site/profile/manifests/nexus/nexus_mem_check.pp]: I tested this on integration with true and false

File: site/profile/manifests/nexus/nexus_mem_check.pp [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    file { '/usr/local/bin/nexus_mem_check.sh':
+      source  => 'puppet:///modules/profile/usr/local/bin/nexus_mem_check.sh',
+      mode    => '0755',
+      owner   => 'root',
+      group   => 'root',
+      require => Package[awscli];
```
[the remark this answers is not in the document]
Comment (arunjadhavt) [reply site/profile/manifests/nexus/nexus_mem_check.pp]: @wanix I don't see any lint warning  for "awscli should be a string in this case"

File: site/profile/manifests/nexus/nexus_mem_check.pp (hunk 1 above)
Comment (arunjadhavt) [reply site/profile/manifests/nexus/nexus_mem_check.pp]: Updated and tested on integration

File: site/profile/manifests/nexus/nexus_mem_check.pp (hunk 2 above)
Comment (arunjadhavt) [reply site/profile/manifests/nexus/nexus_mem_check.pp]: tested  with adding in vagrant hiera. do i need to commit code for vagrant hiera ?

File: site/profile/manifests/nexus/nexus_mem_check.pp (hunk 3 above)
[the remark this answers is not in the document]
Comment (arunjadhavt) [reply site/profile/manifests/nexus/nexus_mem_check.pp]: this is not needed so removed

File: site/profile/manifests/nexus/nexus_mem_check.pp (hunk 2 above)
Comment (arunjadhavt) [reply site/profile/manifests/nexus/nexus_mem_check.pp]: Updated and tested using vagrant hiera and integration stack . updated screen shot of test in jira TICO-901

File: site/profile/manifests/nexus/nexus_mem_check.pp (hunk 3 above)
[the remark this answers is not in the document]
Comment (arunjadhavt) [reply site/profile/manifests/nexus/nexus_mem_check.pp]: not needed here so removed it

**Comment by arunjadhavt** [comment]
@wanix  i have attached rspec test to TICO-901

**Comment by wanix** [comment]
@arunjadhavt , test is completely part of installer code:
https://github.com/Talend/talend-cloud-installer/blob/master/spec/acceptance/shared/nexus.rb

**Comment by arunjadhavt** [comment]
> @arunjadhavt , test is completely part of installer code:
> https://github.com/Talend/talend-cloud-installer/blob/master/spec/acceptance/shared/nexus.rb
> 
> To test it locally: `./scripts/local_dev_tests.sh -t role-nexus-centos-75`

@wanix tested rspec and updated result to TICO-901

**Pull request merged**